src/pe_reports/report_gen/views.py

```diff
@@ -0,0 +1,181 @@
+"""Classes and associated functions that render the UI app pages."""
+
+# Standard Python Libraries
+import datetime
+import logging
+import os
+
+# Third-Party Libraries
+from flask import Blueprint, flash, redirect, render_template, url_for
+
+# cisagov Libraries
+from pe_reports.data.db_query import get_orgs_df
+from pe_reports.helpers.bulletin.bulletin_generator import (
+    generate_creds_bulletin,
+    generate_cybersix_bulletin,
+)
+from pe_reports.report_gen.forms import (
+    BulletinFormExternal,
+    CredsFormExternal,
+    InfoFormExternal,
+)
+from pe_reports.report_generator import generate_reports
+
+LOGGER = logging.getLogger(__name__)
+
+conn = None
+cursor = None
+thedateToday = datetime.date.today().strftime("%Y-%m-%d")
+
+
+report_gen_blueprint = Blueprint(
+    "report_gen", __name__, template_folder="templates/report_gen_UI"
+)
+
+
+def validate_filename(filename):
+    """Verify that a filename is the correct format."""
+    if filename == "":
+        return False
+    if any(
+        char in filename
+        for char in [
+            "#",
+            "%",
+            "&",
+            "{",
+            "}",
+            "<",
+            ">",
+            "!",
+            "`",
+            "$",
+            "+",
+            "*",
+            "'",
+            '"',
+            "?",
+            "=",
+            "/",
+            ":",
+            " ",
+            "@",
+        ]
+    ):
+        return False
+    else:
+        return True
+
+
+def validate_date(date_string):
+    """Validate that a provided string matches the right format and is a report date."""
+    try:
+        date = datetime.datetime.strptime(date_string, "%Y-%m-%d")
+    except ValueError:
+        return False
+    if date.day == 15:
+        return True
+    if date.month in [4, 6, 9, 11] and date.day != 30:
+        return False
+    elif date.month in [1, 3, 5, 7, 8, 12] and date.day != 31:
+        return False
+    elif date.month == 2 and date.day not in [28, 29]:
+        return False
+    return True
+
+
+@report_gen_blueprint.route("/report_gen", methods=["GET", "POST"])
+def report_gen():
+    """Process form information, instantiate form and render page template."""
+    report_date = False
+    output_directory = False
+
+    formExternal = InfoFormExternal()
+
+    if formExternal.validate_on_submit() and formExternal.submit.data:
+        LOGGER.info("Got to the submit validate")
```

Comment:
This seems more like a debug-level message than info, but feel free to disregard if you disagree.
```suggestion
        LOGGER.debug("Got to the submit validate")
```

Reply:
Removed this line in the following commit: [e1a85f5](https://github.com/cisagov/pe-reports/pull/282/commits/e1a85f5fa5b98f4073bf7c09e6abe927d30a30c4)